src/poll.rs

```diff
@@ -141,6 +153,19 @@ impl Poller {
                 FdData {
                     poll_fds_index,
                     key: ev.key,
+                    remove: match mode {
+                        PollMode::Oneshot => true,
+                        PollMode::Level => false,
+                        PollMode::Edge => {
+                            return Err(io::Error::new(
+                                #[cfg(not(polling_no_unsupported_error_kind))]
+                                io::ErrorKind::Unsupported,
+                                #[cfg(polling_no_unsupported_error_kind)]
+                                io::ErrorKind::Other,
```

Comment:
and this should be definitively factored out into something at the top-level (but obviously not publicly exported) to reuse it in all the backends which don't support some case.